pkg/jobstore/envelope.go

```diff
@@ -0,0 +1,63 @@
+package jobstore
+
+import "encoding/json"
+
+// Envelope provides a wrapper around types that can be stored in a jobstore.
+// It takes responsibility for ser/de for the wrapped type, ensuring that
+type Envelope[T any] struct {
+	Body      T
+	marshal   MarshalFunc
+	unmarshal UnmarshalFunc
+}
```

Comment:
Nice. We need to include other metadata to help deserialize the object and maintain backward compatibility. At a minimum we need a `version` field, which will be the version of the envelope. If we decide to include more fields or blocks in the future (e.g. encoding type, object type, object version), we can bump the version and add those later

Reply:
I'll copy the pattern suggestedi n the plugin PR. I wonder whether we should have the Envelope only work for structs that implement a VersionedType struct which could supply the version number as part of it's interface?

Reply:
Yeah having a unified struct or at least a pattern for jobstore and grpc/http requests would be nice. I won't be blocking the PR for this change and we can do it when you are back and before our next release